flink-1.17-connector-bigquery/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/RowDataToProtoSerializerTest.java

```diff
@@ -0,0 +1,517 @@
+        assertEquals(0.5616495161359795, message.getField(descriptor.findFieldByNumber(2)));
+        assertEquals("String", message.getField(descriptor.findFieldByNumber(3)));
+        assertEquals(true, message.getField(descriptor.findFieldByNumber(4)));
+        assertEquals(
+                ByteString.copyFrom(byteArray), message.getField(descriptor.findFieldByNumber(5)));
+        Descriptors.FieldDescriptor fieldDescriptor = descriptor.findFieldByNumber(6);
+        message = (DynamicMessage) message.getField(fieldDescriptor);
+        assertEquals(
+                "hello",
+                message.getField(
+                        descriptor
+                                .findNestedTypeByName(fieldDescriptor.toProto().getTypeName())
+                                .findFieldByNumber(1)));
+    }
+
+    /**
+     * Test to check <code>getDynamicMessageFromRowData()</code> for Primitive types supported by
+     * BigQuery. However <code>null</code> value is passed to the record field to test for error.
+     */
+    @Test
+    public void testAllBigQuerySupportedPrimitiveTypesConversionToDynamicMessageIncorrectly() {
+        // Obtaining the Schema Provider and the Row Data Record.
+        // -- Non-nullable Schema for descriptor
+        BigQuerySchemaProvider bigQuerySchemaProvider =
+                TestBigQuerySchemas.getSchemaWithRequiredPrimitiveTypes();
+        Descriptors.Descriptor descriptor = bigQuerySchemaProvider.getDescriptor();
+        // -- Nullable Schema for descriptor
+        Schema avroSchema =
+                TestBigQuerySchemas.getSchemaWithNullablePrimitiveTypes().getAvroSchema();
+
+        LogicalType logicalType =
+                BigQuerySchemaProviderTableImpl.getDataTypeSchemaFromAvroSchema(avroSchema)
+                        .getLogicalType();
+        RowDataToProtoSerializer rowDataSerializer = new RowDataToProtoSerializer();
+        rowDataSerializer.init(bigQuerySchemaProvider);
+        rowDataSerializer.setLogicalType(logicalType);
+        GenericRowData row = new GenericRowData(6);
+        row.setField(0, null);
+        row.setField(1, null);
+        row.setField(2, null);
+        row.setField(3, null);
+        row.setField(4, null);
+        row.setField(5, null);
+
+        // Check for the desired results.
+        IllegalArgumentException exception =
+                assertThrows(
+                        IllegalArgumentException.class,
+                        () ->
+                                rowDataSerializer.getDynamicMessageFromRowData(
+                                        row, descriptor, logicalType));
+        Assertions.assertThat(exception)
+                .hasMessageContaining("Received null value for non-nullable field");
+    }
+
+    /**
+     * Test to check <code>getDynamicMessageFromRowData()</code> for Primitive types supported by
+     * Avro but not offered by BigQuery.
```

Comment:
just for my own understanding, what does it mean for a primitive type to be supported by avro but not offered by BQ? what are the implications of this? IIUC, the type be cast to a primitive type that is supported by BQ?

Reply:
certain primitive types such as double, fixed are not present in bigquery, but are supported by avro and table api schema. Thought it important to check if these are handled as well.

Reply:
yep makes sense. thank you! you can resolve this convo